virtual_ecosystem/models/animal/animal_model.py

```diff
@@ -513,6 +585,142 @@ def calculate_total_litter_consumption(
+            0,
+        )
+        ectomycorrhiza_initial_stock = where(
+            self.data["soil_c_pool_ectomycorrhiza"].to_numpy() > 0,
+            self.data["soil_c_pool_ectomycorrhiza"].to_numpy(),
+            0,
+        )
+        fungi_initial_stock = (
+            saprotrophic_fungi_initial_stock
+            + arbuscular_mycorrhiza_initial_stock
+            + ectomycorrhiza_initial_stock
+        )
+
+        fungi_final_stock = array(
+            [
+                soil_pools[cid]["fungi"].mass_current
+                / (area * self.core_constants.max_depth_of_microbial_activity)
+                for cid in cell_ids
+            ]
+        )
+        fungi_consumption = fungi_initial_stock - fungi_final_stock
+
+        # Calculate how much of the fungal consumption is applied to each fungal group
+        # based on initial pool sizes
+        saprotrophic_fungi_consumption = fungi_consumption * (
+            saprotrophic_fungi_initial_stock / fungi_initial_stock
+        )
+        ectomycorrhiza_consumption = fungi_consumption * (
+            ectomycorrhiza_initial_stock / fungi_initial_stock
+        )
+        arbuscular_mycorrhiza_consumption = fungi_consumption * (
+            arbuscular_mycorrhiza_initial_stock / fungi_initial_stock
+        )
+
+        return {
+            "animal_pom_consumption_carbon": DataArray(
+                pom_consumption_carbon
+                / self.model_timing.update_interval_quantity.to("days").magnitude,
+                dims="cell_id",
+            ),
+            "animal_pom_consumption_nitrogen": DataArray(
+                pom_consumption_nitrogen
+                / self.model_timing.update_interval_quantity.to("days").magnitude,
+                dims="cell_id",
+            ),
+            "animal_pom_consumption_phosphorus": DataArray(
+                pom_consumption_phosphorus
+                / self.model_timing.update_interval_quantity.to("days").magnitude,
+                dims="cell_id",
+            ),
+            "animal_bacteria_consumption": DataArray(
+                bacteria_consumption
+                / self.model_timing.update_interval_quantity.to("days").magnitude,
+                dims="cell_id",
+            ),
+            "animal_saprotrophic_fungi_consumption": DataArray(
+                saprotrophic_fungi_consumption
+                / self.model_timing.update_interval_quantity.to("days").magnitude,
+                dims="cell_id",
+            ),
+            "animal_ectomycorrhiza_consumption": DataArray(
+                ectomycorrhiza_consumption
+                / self.model_timing.update_interval_quantity.to("days").magnitude,
+                dims="cell_id",
+            ),
+            "animal_arbuscular_mycorrhiza_consumption": DataArray(
+                arbuscular_mycorrhiza_consumption
+                / self.model_timing.update_interval_quantity.to("days").magnitude,
+                dims="cell_id",
+            ),
+        }
```

Comment:
Just for readability, I would use a small helper function:
```suggestion
        days = self.model_timing.update_interval_quantity.to("days").magnitude
        def to_da(array):
            return DataArray(array / days, dims="cell_id")
            
        return {
            "animal_pom_consumption_carbon": to_da(pom_consumption_carbon),
            "animal_pom_consumption_nitrogen": to_da(pom_consumption_nitrogen),
            "animal_pom_consumption_phosphorus": to_da(pom_consumption_phosphorus),
            "animal_bacteria_consumption": to_da(bacteria_consumption),
            "animal_saprotrophic_fungi_consumption": to_da(saprotrophic_fungi_consumption),
            "animal_ectomycorrhiza_consumption": to_da(ectomycorrhiza_consumption),
            "animal_arbuscular_mycorrhiza_consumption": to_da(arbuscular_mycorrhiza_consumption),
        }
```